src/App.jsx

```diff
@@ -14,45 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_NAME = 'name';
+const SORT_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sort, reverse }) {
+  const preparedGoods = [...goods];
+
+  if (sort) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sort) {
+        case SORT_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sort, setSort] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGood = getPreparedGoods(goodsFromServer, { sort, reverse });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(`button is-info`, { 'is-light': sort !== SORT_NAME })}
+          onClick={() => setSort(SORT_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(`button is-success`, { 'is-light': sort !== SORT_LENGTH })}
+          onClick={() => setSort(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', { 'is-light': !reverse })}
+          onClick={() => setReverse(!reverse)}
+        >
+          Reverse
+        </button>
+
+        {(sort || reverse)
+        && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSort('');
+              setReverse(false);
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here